src/main/kotlin/org/rust/ide/inspections/RsDeprecatedAttributeInspection.kt

```diff
@@ -0,0 +1,89 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.inspections
+
+import com.intellij.codeInspection.ProblemHighlightType
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiElementVisitor
+import org.rust.lang.core.psi.RsFile
+import org.rust.lang.core.psi.RsMetaItem
+import org.rust.lang.core.psi.RsOuterAttr
+import org.rust.lang.core.psi.RsVisitor
+import org.rust.lang.core.psi.ext.*
+import org.rust.lang.core.psi.impl.RsModDeclItemImpl
+
+class RsDeprecatedAttributeInspection : RsLocalInspectionTool() {
+    override fun getDisplayName() = "Deprecated item"
+
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor = object : RsVisitor() {
+        override fun visitElement(ref: RsElement) {
+            // non-inline module declaration
+            if (ref is RsModDeclItemImpl) return
+
+            ref.reference?.resolve()?.let {
+                val identifier = if (ref is RsReferenceElement) ref.referenceNameElement else ref
```

Comment:
Let's check that ref is `RsReferenceElement` before name resolution to avoid redundant `resolve` call

Reply:
Fixed